src/controllers/general.controller.js

```diff
@@ -0,0 +1,80 @@
+const Doctors = require("../models/doctor.model");
+const Hospital = require("../models/hospital.model");
+const { searchDoctorSchema } = require("../validators/general.validate");
+
+
+exports.welcome = async (req,res) => {
+    try{
+        return res.send("welcome to site")
+    }catch(err){
+        console.error(`Error`, err.message);
+        next(err);
+    }
+}
+
+exports.hospitalList = async (req,res) =>{
+    try{
+    const hospital = await Hospital.find()
+       return res.json({
+        status : "Success",
+        error : false,
+        message : "Successfully displayed hospital list",
+        data :  hospital
+      })
+    }catch(err){
+        return res.json({
+                            status : "Failed",
+                            error : true,
+                            message : "Failed to display hospital list",
+                            data : { result: err}
+                        })
+    }
+
+}
+
+
+
+exports.searchDoctor = async(req,res) =>{
+    try{
+        const result = await searchDoctorSchema.validateAsync(req.query)
+        let search = result.doctorName
+        let page = 1
+        if(result.page){
+            page = result.page
+        }
+        let limit = 3
+        
+        let doctors = await Doctors.find({doctorName:{$regex:'.*' + search + '.*',$options:'i'}})
```

Comment:
instead of doing `'.*' + search + '.*'` I suggest use string template.

Reply:
Changed [Here](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/3/files#r1045549650)